intellij-plugin-structure/structure-base/src/main/kotlin/com/jetbrains/plugin/structure/base/problems/PluginProblems.kt

```diff
@@ -23,3 +29,36 @@ val PluginProblem.isInvalidDescriptorProblem: Boolean
   } else {
     this is InvalidDescriptorProblem
   }
+
+fun PluginProblem.isInstance(pluginProblemClass: KClass<*>): Boolean =
+  pluginProblemClass.isInstance(unwrapped)
+
+
+private const val PLUGIN_PROBLEM_PACKAGE_DEFAULT_PREFIX = "com.jetbrains.plugin.structure."
+
+object PluginProblems {
+  /**
+   * Resolves the problem ID to a fully qualified Kotlin class.
+   *
+   * The following formats are supported:
+   *
+   * - Fully qualified problem ID which corresponds to a class name, such as
+   *    `com.jetbrains.plugin.structure.intellij.problems.ForbiddenPluginIdPrefix`
+   * - Problem ID which can be resolved to a fully qualified class name in the `com.jetbrains.plugin.structure`
+   *    package prefix, such as `intellij.problems.ForbiddenPluginIdPrefix`.
+   */
+  fun resolveClass(problemId: String): KClass<out Any>? {
+    val fqProblemId = if (problemId.startsWith(PLUGIN_PROBLEM_PACKAGE_DEFAULT_PREFIX)) {
+      problemId
+    } else {
+      PLUGIN_PROBLEM_PACKAGE_DEFAULT_PREFIX + problemId
+    }
+    return runCatching {
+      val pluginProblemJavaClass = Class.forName(fqProblemId, false, this.javaClass.getClassLoader())
+      val kotlin = pluginProblemJavaClass.kotlin
+      kotlin
+    }.onFailure { t ->
+      LOG.warn("Problem ID '$problemId' could not be resolved to a fully qualified class corresponding to a plugin problem: {}", t.message)
```

Comment:
```suggestion
      LOG.warn("Problem ID '$problemId' could not be resolved to a class corresponding to a plugin problem: {}", t.message)
```

If "fully qualified" is actually redundant.

Reply:
Yes, it is redundant. Fixed.